sql3/test/defs/defs_timestamp_literals.go

```diff
@@ -27,11 +29,34 @@ var timestampLiterals = TableTest{
 		{
 			// InsertWithCurrentDate
 			SQLs: sqls(
-				"insert into testtimestampliterals (_id, a, b, d, ts, event, ievent) values (4, 40, 400, 10.12, current_date, ['A', 'B', 'C'], [1, 2, 3])",
+				"insert into testtimestampliterals (_id, a, b, d, ts, event, ievent) values (2, 40, 400, 10.12, current_date, ['A', 'B', 'C'], [1, 2, 3])",
+			),
+			ExpHdrs: hdrs(),
+			ExpRows: rows(),
+			Compare: CompareExactUnordered,
+		},
+		{
+			// Insert literal 0 into a timestamp, it should be stored as 1970-01-01 00:00:00 +0000 UTC (unix epoch base value)
+			SQLs: sqls(
+				"insert into testtimestampliterals (_id, a, b, d, ts, event, ievent) values (3, 40, 400, 10.12, 0, ['A', 'B', 'C'], [1, 2, 3])",
 			),
 			ExpHdrs: hdrs(),
 			ExpRows: rows(),
 			Compare: CompareExactUnordered,
 		},
+		{
+			SQLs: sqls(
+				"select datepart('yy', ts) as \"yy\" from testtimestampliterals",
+			),
+			ExpHdrs: hdrs(
+				hdr("yy", fldTypeInt),
+			),
+			ExpRows: rows(
+				row(int64(1970)),
+				row(int64(time.Now().UTC().Year())),
```

Comment:
okay, i don't fully understand how current_timestamp works, and i know that this will come up extremely rarely if at all, but:

i think it's possible for this test to fail because the computation of `time.Now()` is separate from the computation/handling of `current_timestamp`. So for instance, if we build the table in featurebase, and then a millisecond later we compute our expected rows, that can result in different years. And I recognize that this is very rare, but conceptually, I think we should ensure that the timestamps we're testing against are *identical* to the timestamps that are in the database, not just probably-similar. Even if the probability is > 99.9999%.

Reply:
Comparison test is updated to look for expected year only for those cases where we can pre-determine the year.